tendermint/src/types/proto/info.rs

```diff
@@ -82,12 +82,18 @@ pub struct Evidence {
     /// The height when the offense occurred
     pub height: i64,
     /// The corresponding time where the offense occurred
-    pub time: Option<Timestamp>,
+    pub time: Timestamp,
     /// Total voting power of the validator set in case the ABCI application does
     /// not store historical validators.
     pub total_voting_power: i64,
 }
 
+impl Evidence {
+    pub fn kind(&self) -> EvidenceType {
+        EvidenceType::from_i32(self.r#type).unwrap_or(EvidenceType::Unknown)
```

Comment:
Why not change field to `EvidenceType` instead of this + rename `r#type` to `kind` and use `serde(rename = "type"`